### PR TITLE
create tests

### PR DESCRIPTION
**Pull request by DeborahAlmeida**
## Objective and Motivation
- Make tests to List Battle Endpoint
- Make tests to Create Team Endpoint
- Make tests to Create Battle Endpoint
- Make tests to Detail Battle Endpoint

### CODE REVIEW

File: backend/battle/tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+    def test_login_user_can_acess_battle_list(self):
+        response = self.auth_client.get(reverse('battle-list'))
+        self.assertEqual(response.status_code, 200)
+
+    def test_endpoint_returns_empty_list(self):
+        battle = baker.make('battle.Battle', creator=self.user_1)
+        response = self.auth_client.get(reverse('battle-list'))
+        matching_battles = BattleSerializer([battle], many=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertCountEqual(matching_battles.data, response.json())
```
Comment (rsarai) [inline backend/battle/tests/test_api.py]: Shouldn't this be an empty list?

File: backend/battle/tests/test_api.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+    def test_endpoint_returns_one_battle_in_list(self):
+        battle = baker.make('battle.Battle', creator=self.user_1)
+        response = self.auth_client.get(reverse('battle-list'))
+        matching_battles = BattleSerializer([battle], many=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertCountEqual(matching_battles.data, response.json())
+
+    def test_endpoint_returns_few_battle_ids(self):
+        battle = baker.make('battle.Battle', creator=self.user_1, _quantity=3)
```
Comment (rsarai) [inline backend/battle/tests/test_api.py]: Typo: `battles`

File: backend/battle/tests/test_api.py [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+        response = self.auth_client.get(reverse('battle-list'))
+        matching_battles = BattleSerializer(battle, many=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertCountEqual(matching_battles.data, response.json())
+
+    def test_endpoint_returns_a_lot_battle_ids(self):
+        battle = baker.make('battle.Battle', creator=self.user_1, _quantity=100)
```
Comment (rsarai) [inline backend/battle/tests/test_api.py]: Typo: `battles`

File: backend/battle/tests/test_api.py [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+        self.assertEqual(response.status_code, 200)
+        self.assertCountEqual(matching_battles.data, response.json())
+
+    def test_endpoint_returns_an_error_when_the_user_is_not_logged(self):
+        self.auth_client.logout()
+        response = self.client.get(reverse('battle-list'))
+        self.assertEqual(response.status_code, 403)
+
+    def test_endpoint_diff_to_verify_it_returns_exactly_len_of_battle_list(self):
+        battle = baker.make('battle.Battle', creator=self.user_1, _quantity=10)
```
Comment (rsarai) [inline backend/battle/tests/test_api.py]: Typo: `battles`

File: backend/battle/tests/test_api.py [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+        response_updated = self.auth_client.get(reverse('battle-list'))
+
+        self.assertCountEqual(
+            BattleSerializer(all_battles, many=True).data,
+            response_updated.json())
+
+
+class DeatilBattleEndpointTest(APITestCaseUtils):
+
+    def test_login_user_can_acess_battle_detail(self):
```
Comment (rsarai) [inline backend/battle/tests/test_api.py]: Name suggestion: `test_logged_user_can_acess_battle_detail`

File: backend/battle/tests/test_api.py [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+        battle = baker.make("battle.Battle", creator=self.user_1, opponent=self.opponent)
+        response = self.auth_client.get(reverse('battle-detail', kwargs={'pk': battle.id}))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            BattleSerializer([battle], many=True).data[0]['creator'],
+            response.json()['id'])
```
Comment (rsarai) [inline backend/battle/tests/test_api.py]: This is a bit confusing, I'm not sure what you are testing here. Is that if the creator has the same id of the battle? If yes, this doesn't look like the appropriate thing to test

File: backend/battle/tests/test_api.py [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+        self.assertEqual(response.status_code, 201)
+        self.assertTrue(battle)
+
+    def test_if_returns_error_when_creator_and_opponent_are_the_same_user(self):
+        battle_data = {
+            "creator": self.user_1.id,
+            "opponent": self.user_1.email,
+        }
+        response = self.auth_client.post(reverse('battle-create'), battle_data)
+        # import ipdb; ipdb.set_trace()
```
Comment (rsarai) [inline backend/battle/tests/test_api.py]: :eyes:

**Review by rsarai — CHANGES_REQUESTED** [review]

File: backend/battle/tests/test_api.py (hunk 1 above)
Comment (DeborahAlmeida) [reply backend/battle/tests/test_api.py]: yup! I didn't realize that I was creating a battle, sorry :)

File: backend/battle/tests/test_api.py (hunk 2 above)
Comment (DeborahAlmeida) [reply backend/battle/tests/test_api.py]: yup

File: backend/battle/tests/test_api.py (hunk 3 above)
Comment (DeborahAlmeida) [reply backend/battle/tests/test_api.py]: thanks :)

File: backend/battle/tests/test_api.py (hunk 5 above)
Comment (DeborahAlmeida) [reply backend/battle/tests/test_api.py]: great!

File: backend/battle/tests/test_api.py (hunk 6 above)
Comment (DeborahAlmeida) [reply backend/battle/tests/test_api.py]: It's true, thanks for the advice :)

File: backend/battle/tests/test_api.py (hunk 7 above)
Comment (DeborahAlmeida) [reply backend/battle/tests/test_api.py]: ops! Thank you :D

File: backend/battle/tests/test_api.py [hunk 8]
```diff
@@ -0,0 +1,348 @@
...
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        self.auth_client.post(
+            reverse("team-create"), pokemons_data, follow=True)
+
+        battle_updated = Battle.objects.get(creator=self.user_1, opponent=self.opponent)
```
Comment (rsarai) [inline backend/battle/tests/test_api.py:220]: If you want to get an updated version of the battle you can use: `battle.refresh_from_db()`

**Review by rsarai — APPROVED** [review]

File: backend/battle/tests/test_api.py (hunk 8 above)
Comment (DeborahAlmeida) [reply backend/battle/tests/test_api.py:220]: great!

**Pull request merged**